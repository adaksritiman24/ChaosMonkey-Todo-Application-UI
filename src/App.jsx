import { Grid, Stack, useTheme } from '@mui/material';
import React, { useReducer, useEffect } from 'react'
import Body from './components/Body';
import Header from './components/Header';
import { ACTIONS } from './constants/actions';
import { state , baseURI} from './constants/constants';
import TodoContext from './contexts/TodoContext';
import todoReducer from './reducers/todoReducer';
import axios from "axios";

const App = ()=> {

    const [appState, dispatch] = useReducer(todoReducer,state)
    const todoContextValue = {appState, dispatch};

    console.log("App is rendering: ");
    
    
    const theme = useTheme();
    const desktopView = theme.breakpoints.up("lg");
    const tabletView = theme.breakpoints.up("md");
    const mobileView = theme.breakpoints.up("xs");
    
    useEffect(() => {
        setTimeout(() => {
            const fetchTodoData = async()=>{
                console.log("Fetching todos from database")
                const response = await axios.get(baseURI+"/todo");
                dispatch({
                    type : ACTIONS.SET_TODOS,
                    payload : response.data ? response.data : [],
                })
                dispatch({
                    type : ACTIONS.SET_LOADING,
                    payload : false,
                })
            }
            fetchTodoData();  
        }, 3000);
    
    },[]);
    
    return (
        <TodoContext.Provider value={todoContextValue}>
            <Grid
                sx = {{
                    margin : 0,
                    padding : 0,
                        [desktopView] : { margin : "0rem 15rem"},
                        [tabletView] : { margin : "0rem 3.5rem"},
                        [mobileView] : { margin : "0rem 1rem"},
                }}
            >
                <Stack spacing={2}>
                    <Header/>
                    <Body/>
                </Stack>
            </Grid>
        </TodoContext.Provider>
    )
}

export default App;