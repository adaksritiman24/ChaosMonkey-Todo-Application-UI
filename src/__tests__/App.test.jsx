import {render, screen} from "@testing-library/react";
import App from "../App";

describe("Testing of App component", ()=>{

    it("Should match snapshot", ()=> {
        const { baseElement } = render(<App/>);     
        expect(baseElement).toMatchSnapshot();
        
    });
})