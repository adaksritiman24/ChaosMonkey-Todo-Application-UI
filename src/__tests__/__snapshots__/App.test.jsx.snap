// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing of App component Should match snapshot 1`] = `
<body>
  <div>
    <div
      class="MuiGrid-root css-3nu7wj-MuiGrid-root"
    >
      <div
        class="css-1p5q5e5-MuiStack-root"
      >
        <div
          class="MuiBox-root css-1rxdjlo"
        >
          <h1
            class="MuiTypography-root MuiTypography-body1 css-67bd63-MuiTypography-root"
          >
            Todo Application
          </h1>
          <div
            class="MuiBox-root css-70qvj9"
          >
            <button
              class="MuiButton-root MuiButton-contained MuiButton-containedSecondary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButtonBase-root  css-1bpvkio-MuiButtonBase-root-MuiButton-root"
              data-testid="todo-create-button"
              tabindex="0"
              type="button"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-elhp89-MuiSvgIcon-root"
                data-testid="ListAltIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M19 5v14H5V5h14m1.1-2H3.9c-.5 0-.9.4-.9.9v16.2c0 .4.4.9.9.9h16.2c.4 0 .9-.5.9-.9V3.9c0-.5-.5-.9-.9-.9zM11 7h6v2h-6V7zm0 4h6v2h-6v-2zm0 4h6v2h-6zM7 7h2v2H7zm0 4h2v2H7zm0 4h2v2H7z"
                />
              </svg>
              <p
                class="MuiTypography-root MuiTypography-body1 css-1k3ks9m-MuiTypography-root"
              >
                Create Todo
              </p>
              <span
                class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
              />
            </button>
          </div>
        </div>
        <div
          class="MuiBox-root css-1spr8f0"
        >
          <div
            class="MuiBox-root css-0"
          >
            <div
              class="MuiToggleButtonGroup-root css-1e9i4it-MuiToggleButtonGroup-root"
              role="group"
            >
              <button
                aria-pressed="true"
                class="MuiButtonBase-root MuiToggleButton-root Mui-selected MuiToggleButton-sizeMedium MuiToggleButton-primary MuiToggleButtonGroup-grouped MuiToggleButtonGroup-groupedHorizontal css-1iudn88-MuiButtonBase-root-MuiToggleButton-root"
                tabindex="0"
                type="button"
                value="all"
              >
                All
                <span
                  class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                />
              </button>
              <button
                aria-pressed="false"
                class="MuiButtonBase-root MuiToggleButton-root MuiToggleButton-sizeMedium MuiToggleButton-primary MuiToggleButtonGroup-grouped MuiToggleButtonGroup-groupedHorizontal css-1iudn88-MuiButtonBase-root-MuiToggleButton-root"
                tabindex="0"
                type="button"
                value="completed"
              >
                Completed
                <span
                  class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                />
              </button>
              <button
                aria-pressed="false"
                class="MuiButtonBase-root MuiToggleButton-root MuiToggleButton-sizeMedium MuiToggleButton-primary MuiToggleButtonGroup-grouped MuiToggleButtonGroup-groupedHorizontal css-1iudn88-MuiButtonBase-root-MuiToggleButton-root"
                tabindex="0"
                type="button"
                value="incomplete"
              >
                Incomplete
                <span
                  class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                />
              </button>
            </div>
            <hr />
            <div
              class="MuiBox-root css-1jpm9pd"
            >
              <div
                class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-2 css-mhc70k-MuiGrid-root"
              >
                <div
                  class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-6 MuiGrid-grid-lg-4 css-1tppydz-MuiGrid-root"
                >
                  <div
                    class="MuiBox-root css-1tesdz0"
                  >
                    <div
                      class="css-1nva18r-MuiStack-root"
                    >
                      <span
                        class="MuiSkeleton-root MuiSkeleton-rectangular MuiSkeleton-pulse css-tqb5fd-MuiSkeleton-root"
                        style="width: 80%; height: 40px;"
                      />
                      <span
                        class="MuiSkeleton-root MuiSkeleton-rectangular MuiSkeleton-pulse css-tqb5fd-MuiSkeleton-root"
                        style="height: 95px;"
                      />
                      <div
                        class="css-1y2jz8o-MuiStack-root"
                      >
                        <span
                          class="MuiSkeleton-root MuiSkeleton-rectangular MuiSkeleton-pulse css-flnd2e-MuiSkeleton-root"
                          style="width: 90px; height: 35px;"
                        />
                        <span
                          class="MuiSkeleton-root MuiSkeleton-rectangular MuiSkeleton-pulse css-flnd2e-MuiSkeleton-root"
                          style="width: 90px; height: 35px;"
                        />
                      </div>
                    </div>
                  </div>
                </div>
                <div
                  class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-6 MuiGrid-grid-lg-4 css-1tppydz-MuiGrid-root"
                >
                  <div
                    class="MuiBox-root css-1tesdz0"
                  >
                    <div
                      class="css-1nva18r-MuiStack-root"
                    >
                      <span
                        class="MuiSkeleton-root MuiSkeleton-rectangular MuiSkeleton-pulse css-tqb5fd-MuiSkeleton-root"
                        style="width: 80%; height: 40px;"
                      />
                      <span
                        class="MuiSkeleton-root MuiSkeleton-rectangular MuiSkeleton-pulse css-tqb5fd-MuiSkeleton-root"
                        style="height: 95px;"
                      />
                      <div
                        class="css-1y2jz8o-MuiStack-root"
                      >
                        <span
                          class="MuiSkeleton-root MuiSkeleton-rectangular MuiSkeleton-pulse css-flnd2e-MuiSkeleton-root"
                          style="width: 90px; height: 35px;"
                        />
                        <span
                          class="MuiSkeleton-root MuiSkeleton-rectangular MuiSkeleton-pulse css-flnd2e-MuiSkeleton-root"
                          style="width: 90px; height: 35px;"
                        />
                      </div>
                    </div>
                  </div>
                </div>
                <div
                  class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-6 MuiGrid-grid-lg-4 css-1tppydz-MuiGrid-root"
                >
                  <div
                    class="MuiBox-root css-1tesdz0"
                  >
                    <div
                      class="css-1nva18r-MuiStack-root"
                    >
                      <span
                        class="MuiSkeleton-root MuiSkeleton-rectangular MuiSkeleton-pulse css-tqb5fd-MuiSkeleton-root"
                        style="width: 80%; height: 40px;"
                      />
                      <span
                        class="MuiSkeleton-root MuiSkeleton-rectangular MuiSkeleton-pulse css-tqb5fd-MuiSkeleton-root"
                        style="height: 95px;"
                      />
                      <div
                        class="css-1y2jz8o-MuiStack-root"
                      >
                        <span
                          class="MuiSkeleton-root MuiSkeleton-rectangular MuiSkeleton-pulse css-flnd2e-MuiSkeleton-root"
                          style="width: 90px; height: 35px;"
                        />
                        <span
                          class="MuiSkeleton-root MuiSkeleton-rectangular MuiSkeleton-pulse css-flnd2e-MuiSkeleton-root"
                          style="width: 90px; height: 35px;"
                        />
                      </div>
                    </div>
                  </div>
                </div>
                <div
                  class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-6 MuiGrid-grid-lg-4 css-1tppydz-MuiGrid-root"
                >
                  <div
                    class="MuiBox-root css-1tesdz0"
                  >
                    <div
                      class="css-1nva18r-MuiStack-root"
                    >
                      <span
                        class="MuiSkeleton-root MuiSkeleton-rectangular MuiSkeleton-pulse css-tqb5fd-MuiSkeleton-root"
                        style="width: 80%; height: 40px;"
                      />
                      <span
                        class="MuiSkeleton-root MuiSkeleton-rectangular MuiSkeleton-pulse css-tqb5fd-MuiSkeleton-root"
                        style="height: 95px;"
                      />
                      <div
                        class="css-1y2jz8o-MuiStack-root"
                      >
                        <span
                          class="MuiSkeleton-root MuiSkeleton-rectangular MuiSkeleton-pulse css-flnd2e-MuiSkeleton-root"
                          style="width: 90px; height: 35px;"
                        />
                        <span
                          class="MuiSkeleton-root MuiSkeleton-rectangular MuiSkeleton-pulse css-flnd2e-MuiSkeleton-root"
                          style="width: 90px; height: 35px;"
                        />
                      </div>
                    </div>
                  </div>
                </div>
                <div
                  class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-6 MuiGrid-grid-lg-4 css-1tppydz-MuiGrid-root"
                >
                  <div
                    class="MuiBox-root css-1tesdz0"
                  >
                    <div
                      class="css-1nva18r-MuiStack-root"
                    >
                      <span
                        class="MuiSkeleton-root MuiSkeleton-rectangular MuiSkeleton-pulse css-tqb5fd-MuiSkeleton-root"
                        style="width: 80%; height: 40px;"
                      />
                      <span
                        class="MuiSkeleton-root MuiSkeleton-rectangular MuiSkeleton-pulse css-tqb5fd-MuiSkeleton-root"
                        style="height: 95px;"
                      />
                      <div
                        class="css-1y2jz8o-MuiStack-root"
                      >
                        <span
                          class="MuiSkeleton-root MuiSkeleton-rectangular MuiSkeleton-pulse css-flnd2e-MuiSkeleton-root"
                          style="width: 90px; height: 35px;"
                        />
                        <span
                          class="MuiSkeleton-root MuiSkeleton-rectangular MuiSkeleton-pulse css-flnd2e-MuiSkeleton-root"
                          style="width: 90px; height: 35px;"
                        />
                      </div>
                    </div>
                  </div>
                </div>
                <div
                  class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-6 MuiGrid-grid-lg-4 css-1tppydz-MuiGrid-root"
                >
                  <div
                    class="MuiBox-root css-1tesdz0"
                  >
                    <div
                      class="css-1nva18r-MuiStack-root"
                    >
                      <span
                        class="MuiSkeleton-root MuiSkeleton-rectangular MuiSkeleton-pulse css-tqb5fd-MuiSkeleton-root"
                        style="width: 80%; height: 40px;"
                      />
                      <span
                        class="MuiSkeleton-root MuiSkeleton-rectangular MuiSkeleton-pulse css-tqb5fd-MuiSkeleton-root"
                        style="height: 95px;"
                      />
                      <div
                        class="css-1y2jz8o-MuiStack-root"
                      >
                        <span
                          class="MuiSkeleton-root MuiSkeleton-rectangular MuiSkeleton-pulse css-flnd2e-MuiSkeleton-root"
                          style="width: 90px; height: 35px;"
                        />
                        <span
                          class="MuiSkeleton-root MuiSkeleton-rectangular MuiSkeleton-pulse css-flnd2e-MuiSkeleton-root"
                          style="width: 90px; height: 35px;"
                        />
                      </div>
                    </div>
                  </div>
                </div>
                <div
                  class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-6 MuiGrid-grid-lg-4 css-1tppydz-MuiGrid-root"
                >
                  <div
                    class="MuiBox-root css-1tesdz0"
                  >
                    <div
                      class="css-1nva18r-MuiStack-root"
                    >
                      <span
                        class="MuiSkeleton-root MuiSkeleton-rectangular MuiSkeleton-pulse css-tqb5fd-MuiSkeleton-root"
                        style="width: 80%; height: 40px;"
                      />
                      <span
                        class="MuiSkeleton-root MuiSkeleton-rectangular MuiSkeleton-pulse css-tqb5fd-MuiSkeleton-root"
                        style="height: 95px;"
                      />
                      <div
                        class="css-1y2jz8o-MuiStack-root"
                      >
                        <span
                          class="MuiSkeleton-root MuiSkeleton-rectangular MuiSkeleton-pulse css-flnd2e-MuiSkeleton-root"
                          style="width: 90px; height: 35px;"
                        />
                        <span
                          class="MuiSkeleton-root MuiSkeleton-rectangular MuiSkeleton-pulse css-flnd2e-MuiSkeleton-root"
                          style="width: 90px; height: 35px;"
                        />
                      </div>
                    </div>
                  </div>
                </div>
                <div
                  class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-6 MuiGrid-grid-lg-4 css-1tppydz-MuiGrid-root"
                >
                  <div
                    class="MuiBox-root css-1tesdz0"
                  >
                    <div
                      class="css-1nva18r-MuiStack-root"
                    >
                      <span
                        class="MuiSkeleton-root MuiSkeleton-rectangular MuiSkeleton-pulse css-tqb5fd-MuiSkeleton-root"
                        style="width: 80%; height: 40px;"
                      />
                      <span
                        class="MuiSkeleton-root MuiSkeleton-rectangular MuiSkeleton-pulse css-tqb5fd-MuiSkeleton-root"
                        style="height: 95px;"
                      />
                      <div
                        class="css-1y2jz8o-MuiStack-root"
                      >
                        <span
                          class="MuiSkeleton-root MuiSkeleton-rectangular MuiSkeleton-pulse css-flnd2e-MuiSkeleton-root"
                          style="width: 90px; height: 35px;"
                        />
                        <span
                          class="MuiSkeleton-root MuiSkeleton-rectangular MuiSkeleton-pulse css-flnd2e-MuiSkeleton-root"
                          style="width: 90px; height: 35px;"
                        />
                      </div>
                    </div>
                  </div>
                </div>
                <div
                  class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-6 MuiGrid-grid-lg-4 css-1tppydz-MuiGrid-root"
                >
                  <div
                    class="MuiBox-root css-1tesdz0"
                  >
                    <div
                      class="css-1nva18r-MuiStack-root"
                    >
                      <span
                        class="MuiSkeleton-root MuiSkeleton-rectangular MuiSkeleton-pulse css-tqb5fd-MuiSkeleton-root"
                        style="width: 80%; height: 40px;"
                      />
                      <span
                        class="MuiSkeleton-root MuiSkeleton-rectangular MuiSkeleton-pulse css-tqb5fd-MuiSkeleton-root"
                        style="height: 95px;"
                      />
                      <div
                        class="css-1y2jz8o-MuiStack-root"
                      >
                        <span
                          class="MuiSkeleton-root MuiSkeleton-rectangular MuiSkeleton-pulse css-flnd2e-MuiSkeleton-root"
                          style="width: 90px; height: 35px;"
                        />
                        <span
                          class="MuiSkeleton-root MuiSkeleton-rectangular MuiSkeleton-pulse css-flnd2e-MuiSkeleton-root"
                          style="width: 90px; height: 35px;"
                        />
                      </div>
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</body>
`;
